docs/JsonSchema.md

```diff
@@ -0,0 +1,372 @@
+`style_key`: the key of style, it is customizable.
+`style_value`: the value of style, it is customizable.
+`info`: the config of the style, whether to expand the style when inserting new text around it.
+
+#### MarkEnd
+
+```ts
+{
+    "type": "mark_end"
+}
+```
+
+`type`: `insert`, `delete`, `mark` or `mark_end`.
+
+### Tree
+
+#### Create
+
+```ts
+{
+    "type": "create",
+    "target": string,
+    "parent": string | null,
+    "fractional_index": UInt8Array
+}
+```
+
+- `type`: `create`, `move` or `delete`.
+- `target`: the string format of target `TreeID` moved.
+- `parent`: the string format of `TreeID` or `null`. If it is `null`, the target node will be a root node.
+- `fractional_index`: the fractional index of the target node.
+
+#### Move
+
+```ts
+{
+    "type": "move",
+    "target": string,
+    "parent": string | null,
+    "fractional_index": UInt8Array
+}
+```
+
+- `type`: `create`, `move` or `delete`.
+- `target`: the string format of target `TreeID` moved.
+- `parent`: the string format of `TreeID` or `null`. If it is `null`, the target node will be a root node.
+- `fractional_index`: the fractional index of the target node.
+
+#### Delete
+
+```ts
+{
+    "type": "delete",
+    "target": string
+}
+```
+
+- `type`: `create`, `move` or `delete`.
+- `target`: the string format of target `TreeID` deleted.
+
+### Unknown
+
+To support backward and forward compatibility of encoding, we have an unknown type. When an `Op` with a newly supported Container from a newer version is decoded into the older version, it will be treated as an unknown type in a more general form, such as binary and string. When the new version decodes an unknown `Op`, the newer version of Loro will know its true type and decode correctly.
+
+So there are two kind of unknown format, binary format and json-string format.
+
+#### Binary Unknown
+
+```ts
+{
+    "type": "unknown",
+    "prop": number,
+    "value_type:": "unknown",
+    "value": EncodeValue
+}
+```
+
+- `type`: just an unknown type.
+- `prop`: a property of the encoded op, it's a number.
+- `value_type`: unknown.
+- `value`: common data types used in encoding.
+
+#### Json Unknown
+
+```ts
+{
+    "type": "unknown",
+    "prop": number,
+    "value_type": "json_unknown",
+    "value": string
+}
+```
+
+- `type`: just an unknown type.
+- `prop`: a property of the encoded op, it's a number.
+- `value_type`: json_unknown.
+- `value`: a string json format of `EncodeValue`
+
+## Value
+
+In this section, we will introduction two *Value* in Loro. One is `LoroValue`, it's an enum of data types supported by Loro, such as the value inserted by `List` or `Map`.
+
+The another is `EncodeValue`, it's just used in encoding module for unknown type.
+
+### LoroValue
+
+These are data types supported by Loro and its json format:
+
+- `null`: `null` or `undefined`
+- `Bool`: `true` or `false`
+- `F64`: `number`(float)
+- `I64`: `number`(signed)
```

Comment:
It also could be bigint to avoid overflow, correct?

Reply:
yes